### PR TITLE
Add libc detection function to common's cmake resources

### PR DESCRIPTION
**Pull request by bretambrose**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cmake/AwsCRuntime.cmake [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# Output is "cruntime" on non-Linux platforms.  Output is "glibc" or "musl" on Linux platforms.
+#
+# Intended usage is for managed language CRTs to use this function to build native artifact paths, facilitating
+# support for alternative C runtimes like Musl.
+#
+function(aws_determine_local_c_runtime target)
+    if (CMAKE_SYSTEM_NAME STREQUAL "Linux")
+        execute_process(COMMAND "ldd" "--version" OUTPUT_VARIABLE AWS_LDD_OUTPUT ERROR_VARIABLE AWS_LDD_OUTPUT)
+        string(FIND "${AWS_LDD_OUTPUT}" "musl" AWS_MUSL_INDEX)
+        string(FIND "${AWS_LDD_OUTPUT}" "GLIBC" AWS_GLIBC_INDEX)
```
Comment (graebm) [inline cmake/AwsCRuntime.cmake:14]: just FYI the output on my dev-desktop doesn't have "GLIBC", so it's going to print "Could not determine C runtime"
> ldd (GNU libc) 2.26
> Copyright (C) 2017 Free Software Foundation, Inc.
> This is free software; see the source for copying conditions.  There is NO
> warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
> Written by Roland McGrath and Ulrich Drepper.

Not sure if it's smarter to: try more pattern matches, or don't even bother with explicit checks for "glibc" 🤷‍♀️

File: CMakeLists.txt [hunk 2]
```diff
@@ -279,6 +279,7 @@ list(APPEND EXPORT_MODULES
     "cmake/AwsSIMD.cmake"
     "cmake/AwsFindPackage.cmake"
     "cmake/AwsFeatureTests.cmake"
+    "cmake/AwsCRuntime.cmake"
```
Comment (graebm) [inline CMakeLists.txt:282]: I don't see this function getting called anywhere? Is it called from some dowstream package?

Comment (bretambrose) [reply CMakeLists.txt:282]: Intended for CRT usage really

File: cmake/AwsCRuntime.cmake (hunk 1 above)
Comment (bretambrose) [reply cmake/AwsCRuntime.cmake:14]: Will add GNU I guess

**Review by graebm — APPROVED** [review]

**Pull request merged**